service/src/main/java/org/cbioportal/genome_nexus/service/internal/CancerHotspotServiceImpl.java

```diff
@@ -156,25 +159,29 @@ public List<Hotspot> getHotspotAnnotationsByGenomicLocation(String genomicLocati
         throws VariantAnnotationNotFoundException, VariantAnnotationWebServiceException,
         CancerHotspotsWebServiceException
     {
-        GenomicLocation location = this.notationConverter.parseGenomicLocation(genomicLocation);
-
-        return this.getHotspotAnnotationsByVariant(this.notationConverter.genomicToHgvs(location));
+        VariantAnnotation variantAnnotation = genomicLocationAnnotationService.getAnnotation(genomicLocation);
+        return this.getHotspotAnnotations(variantAnnotation);
     }
 
     @Override
     public List<AggregatedHotspots> getHotspotAnnotationsByGenomicLocations(List<GenomicLocation> genomicLocations)
         throws CancerHotspotsWebServiceException
     {
-        // convert genomic location to hgvs notation (there is always 1-1 mapping)
-        Map<String, GenomicLocation> variantToGenomicLocation = notationConverter.genomicToHgvsMap(genomicLocations);
```

Comment:
what is the reason we can't use this function anymore? Does it make external calls as well?

Reply:
no, it does not make external calls. But it only works with HgvsVariantAnnotationService (no `gn_vep.region.url` set). When we use `RegionVariantAnnotationService`, we need a map from ensembl rest region to genomic location instead (`notationConverter::genomicToEnsemblRestRegion`, not `notationConverter::genomicToHgvs`).

But after that said, we only need this map to trace back and set a genomic location for each result entry.

There must be better way. e.g. a variant annotation already has all necessary information (chromosome, start, end, ref and variant allele) to construct a genomic location instance.

Reply:
@sheridancbio yes, I agree. Having one internal format (a custom class, not a string) and converting and serialising it to another required format of the external call will eliminate quite some code duplication.

Is there an open proposal for that?

Reply:
I guess I'm just sharing a concept rather than actually proposing the code be refactored now to align with that concept (which would be a big job). There is probably more helpful refactoring which could be done, so I don't think this concept should be prioritized. But if an opportunity comes up (such as if we were to do a major refactoring), then maybe it should be considered.